packages/core/handlers/restart.ts

```diff
@@ -9,7 +9,7 @@ const handler = (({ max = 1 }: { max?: number } = { max: 1 }) => {
 		if (running_controllers.size >= max) {
 			const first = running_controllers.values().next();
 			first.value?.abort();
-			running_controllers.delete(first.value);
+			running_controllers.delete(first.value!);
```

Comment:
I think the switcharoo that we did with the tsconfig caused the surface of this problem...but it's really a non problem because you can try to delete `null` or `undefined` and it will just not delete anything.

Reply:
yes but that should probably be reflected in the types no? what's the signature for delete?

Reply:
> yes but that should probably be reflected in the types no? what's the signature for delete?

Nope, i think this is one of the rare instances when TS is more strict than it needs to be

```ts
interface Set<T> {
    /**
     * Appends a new element with a specified value to the end of the Set.
     */
    add(value: T): this;

    clear(): void;
    /**
     * Removes a specified value from the Set.
     * @returns Returns true if an element in the Set existed and has been removed, or false if the element does not exist.
     */
    delete(value: T): boolean;
    /**
     * Executes a provided function once per each value in the Set object, in insertion order.
     */
    forEach(callbackfn: (value: T, value2: T, set: Set<T>) => void, thisArg?: any): void;
    /**
     * @returns a boolean indicating whether an element with the specified value exists in the Set or not.
     */
    has(value: T): boolean;
    /**
     * @returns the number of (unique) elements in Set.
     */
    readonly size: number;
}
```
we could make the `Set` be `Set<AbortController | undefined | null>` but tbf i'm perfectly fine with just asserting the existance.